### PR TITLE
[feat] Implemented GH-Actions for static code analysis

### PR DESCRIPTION
**Pull request by opxdelwin**
**PR Description:**

This PR addresses the issue titled #186. Previously, GitHub Actions were not enabled for automating static Flutter Code Analysis in this project.

**Feature Implemented:**

To resolve this, a pipeline has been established to test relevant commits for code analysis issues. This ensures that code quality is maintained consistently throughout the development process.

Moving forward, any analysis issues identified will be fixed within the main Actions PR itself, ensuring that our codebase remains clean and maintainable.

**Future**
 - [ ] Implement `flutter test` as part of workflow.

Awaiting everyone's valuable feedback.

### CODE REVIEW

File: lib/models/name_value_model.freezed.dart
Comment (ashitaprasad) [inline lib/models/name_value_model.freezed.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: lib/models/name_value_model.g.dart
Comment (ashitaprasad) [inline lib/models/name_value_model.g.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: lib/screens/home_page/editor_pane/details_card/request_pane/request_headers.dart
Comment (ashitaprasad) [inline lib/screens/home_page/editor_pane/details_card/request_pane/request_headers.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: lib/screens/home_page/editor_pane/details_card/request_pane/request_params.dart
Comment (ashitaprasad) [inline lib/screens/home_page/editor_pane/details_card/request_pane/request_params.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: lib/widgets/headerfield.dart
Comment (ashitaprasad) [inline lib/widgets/headerfield.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: lib/widgets/json_previewer.dart
Comment (ashitaprasad) [inline lib/widgets/json_previewer.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: pubspec.lock
Comment (ashitaprasad) [inline pubspec.lock:1]: Remove changes made to this file. It is not relevant to this PR.

File: test/widget_test.dart
Comment (ashitaprasad) [inline test/widget_test.dart:1]: Remove changes made to this file. It is not relevant to this PR.

File: test/widgets/markdown_test.dart
Comment (ashitaprasad) [inline test/widgets/markdown_test.dart:1]: Remove changes made to this file. It is not relevant to this PR.

**Comment by opxdelwin** [comment]
I've reverted all fixes made to the file for workflows to succeed.

**Comment by ashitaprasad** [comment]
LGTM 🚀

**Pull request merged**